File: guessing-game-bdd-atomic/src/test/java/tw/waterballsa/bddworkshop/stepdefs/hooks/DatabaseCleanupHooks.java
package tw.waterballsa.bddworkshop.stepdefs.hooks;

import io.cucumber.java.Before;
import org.springframework.beans.factory.annotation.Autowired;
import tw.waterballsa.bddworkshop.repository.GreetingRepository;

public class DatabaseCleanupHooks {

    @Autowired
    private GreetingRepository greetingRepository;
    

    @Before
    public void cleanupAllData() {
        // Clean up all repository data before each scenario, regardless of feature or tag
        greetingRepository.deleteAll();
    }
} 

File: guessing-game-bdd-atomic/src/test/stepdefs/hooks/DatabaseCleanupHooks.ts
import { Before } from '@cucumber/cucumber';
import { INestApplication } from '@nestjs/common';
import { Repository } from 'typeorm';
import { Greeting } from '../../../main/greeting/entities/greeting.entity';

export class DatabaseCleanupHooks {
    constructor(
        private readonly app: INestApplication
    ) {}

    @Before()
    public async cleanupAllData(): Promise<void> {
        const greetingRepository = this.app.get<Repository<Greeting>>('GreetingRepository');
        // Clean up all repository data before each scenario, regardless of feature or tag
        await greetingRepository.clear();
    }
}
